src/libaktualizr/primary/aktualizr.cc

```diff
@@ -44,6 +44,7 @@ int Aktualizr::Run() {
 }
 
 void Aktualizr::AddSecondary(const std::shared_ptr<Uptane::SecondaryInterface> &secondary) {
+  secondary->events_channel = events_channel_;
```

Comment:
It feels a bit confusing that this events channel is sometimes passed through the constuctor and sometimes set later (with `nullptr` passed to the constructor). Can it made it so the constructor never sets the channel and it's always set up afterwards, possibly through a dedicated method?

Reply:
Your suggestion conflicts with Antons. I think we should disallow to pass nullptr to the constructor, but for this case we would need to add new API method getEventsChannel()